core/controllers/editor.py

```diff
@@ -404,24 +400,20 @@ def put(self, exploration_id):
                     'recipient.')
             email_manager.require_moderator_email_prereqs_are_satisfied()
 
-        # Perform the moderator action.
-        if action == 'unpublish_exploration':
-            rights_manager.unpublish_exploration(self.user, exploration_id)
-            search_services.delete_explorations_from_search_index([
-                exploration_id])
-        else:
-            raise self.InvalidInputException(
-                'No change was made to this exploration.')
-
+        # Unpublish exploration.
+        rights_manager.unpublish_exploration(self.user, exploration_id)
+        search_services.delete_explorations_from_search_index([
+            exploration_id])
         exp_rights = rights_manager.get_exploration_rights(exploration_id)
 
         # If moderator emails can be sent, send an email to the all owners of
         # the exploration notifying them of the change.
         if feconf.REQUIRE_EMAIL_ON_MODERATOR_ACTION:
+            intent = 'unpublish_exploration'
             for owner_id in exp_rights.owner_ids:
                 email_manager.send_moderator_action_email(
                     self.user_id, owner_id,
-                    feconf.VALID_MODERATOR_ACTIONS[action]['email_intent'],
+                    feconf.VALID_MODERATOR_ACTIONS[intent]['email_intent'],
```

Comment:
Not sure we even need this constant. Why don't we just get rid of it and inline all the  stuff in it?

Reply:
the dictionary VALID_MODERATOR_ACTIONS also store the email config string for config domain which can't be done away with.

Should I instead remove the VALID_MODERATOR_ACTIONS and keep the entry MODERATOR_ACTION_UNPUBLISH_EXPLORATION separately?
![image](https://user-images.githubusercontent.com/9693472/31698238-10e85fd0-b3da-11e7-90ee-0fe200970b02.png)

Reply:
I think you can just get the config property directly, right? I.e. as email_manager.UNPUBLISH_EXPLORATION_EMAIL_HTML_BODY.value?